jsco4180Sup/jsco4180.db

```diff
@@ -1006,18 +742,51 @@ record(bo, "$(P)ERROR:SP")
 
 alias("$(P)ERROR:SP", "$(P)ERROR:SP:RBV")
 
-record(stringin, "$(P)ERROR:STR")
+record(stringin, "$(P)ERROR:STR") 
 {
-    field(DESC, "Command input return error")
+    field(DESC, "Last error returned on input")
     field(DTYP, "stream")
     field(SCAN, "I/O Intr")
-    field(INP, "@jsco4180.proto readErrorString $(PORT)")
+    field(VAL, "No error")
+    
+    field(INP, "@jsco4180.proto readErrorStr $(PORT)")
 
+    info(archive, "VAL")
     field(SIML, "$(P)SIM")
     field(SIOL, "$(P)SIM:ERROR:STR")
     field(SDIS, "$(P)DISABLE")
 }
 
+record(bi, "$(P)ERROR:COMP") 
+{
+    field(DESC, "Device component error")
+    field(DTYP, "Soft Channel")
+    field(VAL, "0")
+    
+    field(ZNAM, "No issue")
+    
+    field(ONAM, "Single Channel")
+    field(OSV, "MAJOR")
+
+    info(archive, "VAL")
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:ERROR:COMP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(ai, "$(P)ERROR:DELAY")
+{
+    field(DESC, "Delay before trying to reset components")
+    field(VAL, "1")
+}
+
+record(ai, "$(P)ERROR:TOL")
+{
+    field(DESC, "Tolerance between comp SP and RBV")
+    field(VAL, "0")
```

Comment:
I feel like this should be set small but non-zero, as the device will always have precision issues at some level.

This would happen if a user tries to set a composition of, for example, `1/3`. The device will respond with finite precision and it will be detected as "out of tolerance" causing it to retry. Likely a tolerance of a few decimal places would be plenty (consult with inst. scis. if not sure).

Reply:
I think you may also need a `field(PINI, "YES")` for this to actually take effect

Reply:
You're right. I've updated the records to have a sensible default and included a `PINI` field.